src/react-components/ory/helpers/filter-flow-nodes.tsx

```diff
@@ -25,20 +25,28 @@ export const FilterFlowNodes = ({
 
   const nodes = filterNodesByGroups(filter)
     // we don't want to map the csrf token every time, only on the form level
-    .filter((node) =>
-      getInputName(node) === "csrf_token" && !includeCSRF ? false : true,
-    )
-    .map((node, k) =>
-      ["hidden"].includes(getNodeInputType(node.attributes))
-        ? {
-            node: <Node node={node} key={k} {...overrides} />,
-            hidden: true,
+    .filter((node) => includeCSRF || !(getInputName(node) === "csrf_token"))
+    .map((node, k) => ({
+      node: (
+        <Node
+          node={node}
+          key={
+            // input node
+            "name" in node.attributes
```

Comment:
don't we have TypeGuards for this? I believe we already use ory/integrations here, so might as well use the TypeGuards? 

https://github.com/ory/integrations/blob/main/src/ui/index.ts#L87

https://github.com/ory/integrations/blob/main/src/ui/index.ts#L54

Reply:
Good call :+1: